fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -3570,182 +4642,260 @@ void analyzerPerField() {
-
-            // Synonym analyzer is chosen due to the keyword "synonym" from the text
-            recordStore.saveRecord(createSimpleDocument(1623L, "synonym food", 1));
-            assertEquals(1, recordStore.scanIndex(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, fullTextSearch(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, "nutrient"), null, ScanProperties.FORWARD_SCAN)
+            if (isSynthetic) {
+                openRecordStore(context, metaDataBuilder -> metaDataHookSyntheticRecord1(metaDataBuilder, index));
+                // Synonym analyzer is chosen due to the keyword "synonym" from the text
+                createSyntheticRecord1(1, 1623L, 1623L, "synonym food", "", false, System.currentTimeMillis(), 0);
+                // Ngram analyzer is chosen due to no keyword "synonym" from the text
+                createSyntheticRecord1(2, 1624L, 1624L, "ngram motivation", "", false, System.currentTimeMillis(), 1);
+            } else {
+                rebuildIndexMetaData(context, SIMPLE_DOC, index);
+                // Synonym analyzer is chosen due to the keyword "synonym" from the text
+                recordStore.saveRecord(createSimpleDocument(1623L, "synonym food", 1));
+                // Ngram analyzer is chosen due to no keyword "synonym" from the text
+                recordStore.saveRecord(createSimpleDocument(1624L, "ngram motivation", 1));
+            }
+            assertEquals(1, recordStore.scanIndex(index, fullTextSearch(index, "nutrient"), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
-            assertEquals(0, recordStore.scanIndex(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, fullTextSearch(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, "foo"), null, ScanProperties.FORWARD_SCAN)
+            assertEquals(0, recordStore.scanIndex(index, fullTextSearch(index, "foo"), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
-
-            // Ngram analyzer is chosen due to no keyword "synonym" from the text
-            recordStore.saveRecord(createSimpleDocument(1624L, "ngram motivation", 1));
-            assertEquals(0, recordStore.scanIndex(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, fullTextSearch(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, "need"), null, ScanProperties.FORWARD_SCAN)
+            assertEquals(0, recordStore.scanIndex(index, fullTextSearch(index, "need"), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
-            assertEquals(1, recordStore.scanIndex(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, fullTextSearch(ANALYZER_CHOOSER_TEST_LUCENE_INDEX, "motivatio"), null, ScanProperties.FORWARD_SCAN)
+            assertEquals(1, recordStore.scanIndex(index, fullTextSearch(index, "motivatio"), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
         }
     }
 
-    @Test
-    void basicLuceneCursorTest() {
+    @ParameterizedTest
+    @MethodSource(LUCENE_INDEX_MAP_PARAMS)
+    void basicLuceneCursorTest(Map<String, Index> indexMap, boolean isSynthetic) {
```

Comment:
I feel like I saw a bunch of tests above about limits/skips that heavily overlap with these, and I'm not sure both sets are necessary.